In [1]:
%matplotlib inline
import numpy as np

import matplotlib.pyplot as plt
import cv2
import seaborn as sns

import pandas as pd

> Les leucémies sont caractérisées par des cellules se trouvant dans le sang qui normalement ne doivent se trouver que dans la moëlle osseuse
>
> On identifie ici les blast, promyelocyte, myelocyte et metamyelocyte.
>
> La catégorie IG regroupe les promyelocyte, myelocyte et metamyelocyte
>
> Certaines cellules sont également non identifiables car "éclatées" lors du frottis, elles sont classées en "smudge"

> L'intérêt est donc de pouvoir identifier ces différentes cellules pour mettre une alerte lors des analyses si ces types de cellules sont présentes
>
> En prenant les images provenant de la base des données leucémiques, on retrouve les catégories précitées
>
> L'idée est donc de classifier les 10 types de cellules
> * smudge, blast, eosinophil, erythroblast, lymphocyte, ig, neutrophil, monocyte, basophil, platelet

In [2]:
data_mendeley = pd.read_csv('../base_mendeley.csv',index_col=0)
data_mendeley.head()

file_path       filename     class  \
0  ../images_mendeley/PBC_dataset_normal_DIB/baso...  BA_100102.jpg  basophil   
1  ../images_mendeley/PBC_dataset_normal_DIB/baso...  BA_101381.jpg  basophil   
2  ../images_mendeley/PBC_dataset_normal_DIB/baso...  BA_101611.jpg  basophil   
3  ../images_mendeley/PBC_dataset_normal_DIB/baso...  BA_102039.jpg  basophil   
4  ../images_mendeley/PBC_dataset_normal_DIB/baso...  BA_102750.jpg  basophil   

      size  
0  363x360  
1  363x360  
2  363x360  
3  363x360  
4  363x360

In [3]:
data_apl_aml = pd.read_csv('../base_apl_aml.csv',index_col=0)
data_apl_aml.head()

Patient_ID                                          file_path  \
0  Patient_00  ../images_apl_aml/All/All/Patient_00/Signed sl...   
1  Patient_00  ../images_apl_aml/All/All/Patient_00/Signed sl...   
2  Patient_00  ../images_apl_aml/All/All/Patient_00/Signed sl...   
3  Patient_00  ../images_apl_aml/All/All/Patient_00/Signed sl...   
4  Patient_00  ../images_apl_aml/All/All/Patient_00/Signed sl...   

          filename  class           size  
0  BL_15336348.jpg  blast  (363, 360, 3)  
1  BL_15336349.jpg  blast  (363, 360, 3)  
2  BL_15336361.jpg  blast  (363, 360, 3)  
3  BL_15336369.jpg  blast  (363, 360, 3)  
4  BL_15336376.jpg  blast  (363, 360, 3)

In [7]:
data = pd.concat([data_mendeley.drop(columns=['size']),data_apl_aml.drop(columns=['size','Patient_ID'])])
data = data.reset_index().drop(columns=['index'])
data.head()

file_path       filename     class
0  ../images_mendeley/PBC_dataset_normal_DIB/baso...  BA_100102.jpg  basophil
1  ../images_mendeley/PBC_dataset_normal_DIB/baso...  BA_101381.jpg  basophil
2  ../images_mendeley/PBC_dataset_normal_DIB/baso...  BA_101611.jpg  basophil
3  ../images_mendeley/PBC_dataset_normal_DIB/baso...  BA_102039.jpg  basophil
4  ../images_mendeley/PBC_dataset_normal_DIB/baso...  BA_102750.jpg  basophil

In [8]:
data['class'].value_counts()

neutrophil      5400
lymphocyte      4618
ig              3815
blast           3296
eosinophil      3225
smudge          2909
monocyte        2719
platelet        2414
erythroblast    2068
basophil        1271
Name: class, dtype: int64

In [9]:
data.to_csv('../base_10_class.csv')

In [10]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 31735 entries, 0 to 31734
Data columns (total 3 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   file_path  31735 non-null  object
 1   filename   31735 non-null  object
 2   class      31735 non-null  object
dtypes: object(3)
memory usage: 743.9+ KB


> **Nous avons désormais une base de 31735 images classifiées en 10 catégories**